nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-documentation/src/main/java/org/apache/nifi/documentation/DocGenerator.java

```diff
@@ -69,25 +70,34 @@ public static void generate(final NiFiProperties properties, final ExtensionMana
     /**
      * Documents a type of configurable component.
      *
-     * @param extensionClasses types of a configurable component
+     * @param extensionDefinitions definitions of the extensions to document
      * @param explodedNiFiDocsDir base directory of component documentation
      */
-    public static void documentConfigurableComponent(final Set<Class> extensionClasses, final File explodedNiFiDocsDir, final ExtensionManager extensionManager) {
-        for (final Class<?> extensionClass : extensionClasses) {
-            if (ConfigurableComponent.class.isAssignableFrom(extensionClass)) {
-                final String extensionClassName = extensionClass.getCanonicalName();
-
-                final Bundle bundle = extensionManager.getBundle(extensionClass.getClassLoader());
-                if (bundle == null) {
-                    logger.warn("No coordinate found for {}, skipping...", new Object[] {extensionClassName});
-                    continue;
-                }
-                final BundleCoordinate coordinate = bundle.getBundleDetails().getCoordinate();
+    public static void documentConfigurableComponent(final Set<ExtensionDefinition> extensionDefinitions, final File explodedNiFiDocsDir, final ExtensionManager extensionManager) {
+        for (final ExtensionDefinition extensionDefinition : extensionDefinitions) {
+            final Bundle bundle = extensionDefinition.getBundle();
+            if (bundle == null) {
+                logger.warn("Cannot document extension {} because it has no bundle associated with it", extensionDefinition);
+                continue;
+            }
+
+            final BundleCoordinate coordinate = bundle.getBundleDetails().getCoordinate();
+
+            final String extensionClassName = extensionDefinition.getImplementationClassName();
+            final String path = coordinate.getGroup() + "/" + coordinate.getId() + "/" + coordinate.getVersion() + "/" + extensionClassName;
+            final File componentDirectory = new File(explodedNiFiDocsDir, path);
+            final File indexHtml = new File(componentDirectory, "index.html");
```

Comment:
It looks like the `index.html` filename string is referenced multiple times in this class.  Is it worth creating a static variable to define it once and reuse it here as well as in the `document()` method?

Reply:
Thanks for the reply, that sounds reasonable.